app/src/main/java/org/oppia/android/app/richtext/RteTextView.kt

```diff
@@ -0,0 +1,428 @@
+      val styledHtmlText = htmlParser?.parseOppiaRichText(
+        context, text?.toString() ?: "", supportsConceptCards
+      ) ?: ""
+      super.setText(styledHtmlText, type)
+      boundRawHtml = text // Save for later rebinds.
+
+      // After the text is set & before re-layout, make sure all deferred drawables are observed so
+      // that, when they're loaded, the text view is re-laid out to fit the newly loaded image. Note
+      // that this operation has some room for performance improvements: it triggers a full
+      // re-layout of the TextView for quickly subsequent loads. It may be preferable in the future
+      // to batch load all assets when they're guaranteed to be stored locally and do a single
+      // layout to avoid flickering or perceived latency.
+      val needsToLayOut = getImageDrawables().fold(initial = false) { needsToReLayOut, drawable ->
+        if (drawable is DeferredImageDrawable<*>) {
+          // Only listen for the drawable if it isn't already loaded.
+          val loadedDrawable = drawable.getLoadedDrawable()
+          if (loadedDrawable.value == null) {
+            viewLifecycleOwner.observeOnce(loadedDrawable) {
+              // This drawable needs to be re-laid out.
+              drawablesRequireLayOut = true
+
+              // The drawable being finished loading requires another re-layout since the layout
+              // step is responsible for fitting images within the text view.
+              requestLayout()
+            }
+          } else {
+            return@fold true // The image is already available. Ensure it's measured immediately.
+          }
+        }
+        return@fold needsToReLayOut
+      }
+      if (needsToLayOut) {
+        // There are drawables that are immediately available for laying out the view.
+        drawablesRequireLayOut = true
+        requestLayout()
+      }
+    }  else {
+      // Trigger internal updates by setting the same text again. This may happen during layout and
+      // is not guaranteed to trigger a re-layout or re-draw.
+      super.setText(currentText, type)
+    }
+  }
+
+  override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
+    super.onMeasure(widthMeasureSpec, heightMeasureSpec)
+
+    val viewWidth = width
+    if (drawablesRequireLayOut || viewWidth != lastViewWidthUsedForDrawables) {
+      layOutImageDrawables(getImageDrawables(), viewWidth)
+      drawablesRequireLayOut = false
+      lastViewWidthUsedForDrawables = viewWidth
+
+      // Re-measure the text view in case the drawables lead to the text view taking up more/less
+      // space than before. Note that only height is expected to ever change since the drawables are
+      // constrained based on the view's computed width.
+      super.onMeasure(widthMeasureSpec, heightMeasureSpec)
+    }
+  }
+
+  private fun layOutImageDrawables(drawables: List<Drawable>, viewWidth: Int) {
```

Comment:
`layOutImageDrawables` or it should be `layoutImageDrawables`?

Reply:
lay out is the verb phrase versus layout which is the noun. Android got this wrong in its naming. :) But, given the inconsistency, I may try to find an alternative name altogether.